### DISCUSSION TITLE
Easier ways to download bottlerocket image for bare metal

### DISCUSSION THREAD

**Discussion by StefanTheWiz** [bottlerocket-os/bottlerocket#2779] [General]
Hi.

According to [this guide](https://github.com/bottlerocket-os/bottlerocket/blob/develop/PROVISIONING-METAL.md) on provisioning for bare metal, to download the OS image, I'd have to install rust then tuftool and a bunch of other stuff to get these working.

Is there a reason why this involves so many steps? 
Could it be simplified down to:
```sh
wget "https://updates.bottlerocket.aws/targets/bottlerocket-${VARIANT}-${ARCH}-${VERSION}.img.lz4"
```

For context: I'm trying to automate some of the steps in the bare metal guide.

**Comment by stockholmux** [comment]
@StefanTheWiz When you say "I'd have to install rust then tuftool and a bunch of other stuff" - what is the "other stuff" you're referring to?

**Reply by StefanTheWiz** [reply]
```sh
apt install -y --no-install-recommends \
    ca-certificates \
    build-essential \
    libssl-dev \
    curl \
    pkg-config
```

**Reply by StefanTheWiz** [reply]
I'm using hashicorp packer to automate building the OS image.
I'm starting with an ubuntu VM in rescue mode so I can write the final image over the block device that will snapshotted.
However, because the VM is in rescue mode, it has limited storage and I start running out of space when installing tuftool.

If the download process was as simple as a link or a bash script, this would reduce my build time down to under 2 mins and it would have saved me about a day's work trying to get this stuff to build and install.

**Reply by zmrow** [reply]
@StefanTheWiz Could you elaborate a bit more on what you're trying to do?  It sounds like you're trying to build an image, rather than download it? 

Building Bottlerocket requires the tools you mentioned (build-essential, curl, etc), but `tuftool` should not.

**Reply by StefanTheWiz** [reply]
I think these came up as part of installing rust if I remember correctly

**Comment by zmrow** [comment]
@StefanTheWiz would it be sufficient to provide binary releases of `tuftool` on Github?  That would avoid needing Rust or the tools you mentioned.

**Reply by StefanTheWiz** [reply]
yes, I think so.

**Comment by tommie** [comment]
I agree that `tuftool` is a really strange requirement for "just downloading" the image. It's very complicated.

I made this script for using `wget` and `jq`: https://gist.github.com/tommie/f4528a7da1af9d8228c385e175462faa

I'd probably also be fine with a `tuftool` binary; something I can easily wget to then fetch the image.

**Comment by tommie** [comment]
I should also add: why do I have to pass three URLs/paths to tuftool? Can't the metadata/targets URL be stored in `root.json`?

**Comment by cbgbt** [comment]
I've created https://github.com/awslabs/tough/issues/662 to track creating binary releases of `tuftool`on GitHub.

**Comment by stockholmux** [comment]
FYI - I added a Dockerfile for `tuftool` which simplifies getting the dependencies and building the tool.
